components/merino/src/curated_recommendations/http.rs

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use super::error::Error;
+use super::models::{CuratedRecommendationsRequest, CuratedRecommendationsResponse};
+use url::Url;
+use viaduct::{header_names, Request, Response};
+
+pub struct HttpClient {}
+
+impl HttpClient {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn make_curated_recommendation_request(
+        &self,
+        request: &CuratedRecommendationsRequest,
+        user_agent_header: &str,
+        base_host: &str, // e.g. "https://merino.services.mozilla.com"
+    ) -> Result<CuratedRecommendationsResponse, Error> {
+        let full_url = format!("{}/api/v1/curated-recommendations", base_host);
+        let url = Url::parse(&full_url)?;
+        log::trace!("making request: {url}");
+        let response: Response = Request::post(url)
+            .header(header_names::ACCEPT, "application/json")?
```

Comment:
If it's supported in viaduct, we can also do gzip.
```suggestion
            .header(header_names::ACCEPT, "application/json")?
            .header(header_names:ACCEPT_ENCODING, "gzip")?
```